src/audio/selector/selector.c

```diff
@@ -285,8 +285,8 @@ static int selector_ctrl_get_data(struct comp_dev *dev,
 			return -EINVAL;
 
 		/* Copy back to user space */
-		ret = memcpy_s(cdata->data->data, ((struct sof_abi_hdr *)
-			       (cdata->data))->size, &cd->config,
+		ret = memcpy_s(cdata->data->data,
+			       size, &cd->config,
```

Comment:
should `((struct sof_abi_hdr *)(cdata->data))->size` be checked for validity?

Reply:
I don't think so given we overwrite it below as the reply.

Reply:
but it comes from the user, right? So I thought that maybe we shouldn't be talking to misbehaving users, or at least should warn the system?

Reply:
In theory yes, but I wonder if it's worth the code on DSP side given there are checks here for "size" and aother check in "ipc_comp_value()". This would be really only useful for a legitimate host that is somehow misconfigured, but that will show up already in logs as the header size will be changed by DSP in the response.

Reply:
Also if we were wanting to change that, it should be a layer above, we should not be doing that check at every module